autoload/go/path.vim

```diff
@@ -134,8 +128,8 @@ function! go#path#BinPath() abort
   elseif $GOBIN != ""
     let bin_path = $GOBIN
   else
-    " GOPATH (user set or default GO)
-    let bin_path = expand(go#path#Default() . "/bin/")
+    let go_paths = split(go#path#Default(), go#util#PathListSep())
```

Comment:
@fatih very much a drive-by comment... i.e. I haven't spent a huge amount of time looking into the context.

But can you not use `go list` to find the location of a binary?

```bash
go list -f '{{.Dir}}' github.com/nsf/gocode
```

That way you can avoid any assumptions about which part of `GOPATH` to use...